src/realm/object-store/sync/async_open_task.cpp

```diff
@@ -66,7 +67,35 @@ void AsyncOpenTask::start(util::UniqueFunction<void(ThreadSafeReference, std::ex
         catch (...) {
             return callback({}, std::current_exception());
         }
-        callback(std::move(realm), nullptr);
+
+        // TODO: add a config param to guard this
+        // before the to
+        // the file has been downloaded.
+        SharedRealm shared_realm = Realm::get_shared_realm(std::move(realm));
+        auto config = shared_realm->config();
+
+        if (config.sync_config && config.sync_config->flx_sync_requested) {
+            using namespace sync;
+            // I am assuming that the latest subscription at this point is running the initial query
+            // Am I wrong?
+            auto subscription = shared_realm->get_latest_subscription_set();
```

Comment:
The goal is to have an "init subscription" callback that triggers before we signal that the download from the server has been completed.
I am assuming that the latest subscription at this stage is running the initial query, thus I am registering a callback in order to be notified and then invoke the other callbacks afterwards.

Reply:
Isn't the idea to have the option to pass initial subscriptions? The callback should be registered to be notified when that one completes bootstrapping.